test/exfmt/integration/def_test.exs

```diff
@@ -11,14 +11,12 @@ defmodule Exfmt.Integration.DefTest do
   end
 
   test "def with long guard" do
-    """
-    def one?(x) when x in [:one, "one", 1, "1"] do
-      true
-    end
-    """ ~> """
-    def one?(x)
-        when x in [:one, "one", 1, "1"] do
-      true
+    assert_format """
+    defp valid?(left, doc, right)
+         when is_doc(left)
+         when is_doc(doc)
+         when is_doc(right) do
```

Comment:
Funny, but it is valid. Multiple ```when``` expressions are implicitly ```or```ed together. This test case is very similar to the example given in the style guide section on the formatting guards.

I learned this and few more surprising facts while [researching guards](https://github.com/elixir-lang/elixir/blob/master/lib/elixir/pages/Guards.md#multiple-guards-in-the-same-clause). For example, there's a subtle difference in the case of failing guards when an explicit ```or``` isn't given. 

> In case of a boolean expression guard, a failed element means the whole guard fails. In case of multiple guards it means the next one will be evaluated.